### DISCUSSION TITLE
about http: multiple registrations for /metrics error

### DISCUSSION THREAD

**Discussion by Autumn-27** [projectdiscovery/nuclei#6581] [General]
ne, err := nuclei.NewNucleiEngine(
		nuclei.WithTemplateFilters(nuclei.TemplateFilters{Tags: []string{"oast"}}),
		nuclei.EnableStatsWithOpts(nuclei.StatsOptions{MetricServerPort: 6064}), // optionally enable metrics server for better observability
	)

How to close the MetricServerPort here? It seems impossible to directly call certain switches from the code

**Comment by Mzack9999** [comment]
Would you like it to be able to close it with a cancel function after the settings has been provided and nuclei SDK is running?

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
